src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +32,33 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{todo.id}
             </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={onClose}
+              aria-label="Close modal"
```

Comment:
The `aria-label` for the close button is well-implemented, providing accessibility for screen readers. Ensure all interactive elements maintain this level of accessibility.